example/spec/helpers/kinesisHelpers.js

```diff
@@ -159,8 +161,15 @@ async function getShardIterator(streamName) {
  *                                  Shard iterators must be generated using getShardIterator.
  * @returns {Promise}              - kinesis GetRecords promise
  */
-async function getRecords(shardIterator) {
-  return kinesis.getRecords({ ShardIterator: shardIterator }).promise();
+async function getRecords(shardIterator, records = []) {
+  const data = await kinesis.getRecords({ ShardIterator: shardIterator }).promise();
+  records.push(...data.Records);
+  if ((data.NextShardIterator !== null) &&
+      (data.Records.length === 0 || data.MillisBehindLatest > 0)) {
+    await timeout(waitPeriodMs);
+    return getRecords(data.NextShardIterator, records);
+  }
```

Comment:
Call get records in a loop if we didn't get any last time (length = 0), or we aren't caught up to latests (millisbehindLatest) https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Kinesis.html#getRecords-property